apps/functions-subscription/src/domain/subscription.ts

```diff
@@ -0,0 +1,44 @@
+import { pipe } from 'fp-ts/lib/function';
+import * as O from 'fp-ts/Option';
+import * as RTE from 'fp-ts/ReaderTaskEither';
+import * as TE from 'fp-ts/TaskEither';
+import { Capabilities } from './capabilities';
+
+export type SubscriptionId = string & { readonly __tag: unique symbol };
+export type UserId = string & { readonly __tag: unique symbol };
+export type TrialId = string & { readonly __tag: unique symbol };
+
+export interface Subscription {
+  readonly id: SubscriptionId;
+  readonly userId: UserId;
+  readonly trialId: TrialId;
+  readonly activatedAt?: Date;
+}
```

Comment:
**note**: These types can be defined using `io-ts`. However, I haven't done so because the system doesn't currently require them. We can define them later if needed.